internal/client/project.go

```diff
@@ -111,6 +113,18 @@ func New(ctx context.Context, opts ...Option) (*Project, error) {
 	return client, nil
 }
 
+// addMetadata adds any grpc metadata that should be include by any gRPC
+// calls that are done with returned context. This is used to inject metadata
+// about the local runner, allow the server to target jobs back to the client
+// that performed an RPC call.
+func (c *Project) AddMetadata(ctx context.Context) context.Context {
```

Comment:
This is the only thing that's kind of suspect to me.

It feels like maybe we should do a few things instead: 

* Have an exported func on `Project` that is like `TargetRunnerId` which would return non-empty if one exists (active runner exists basically).
* Add a `WithRunnerId` func to `internal/server` so we have the yin/yang of inject/extract for context metadata.
* Have `internal/cli` do a `if TargetRunnerId() != "" { ctx = server.WithRunnerId(...) }` type thing.

I agree with what you're trying to do here, just not sure if organizationally/structurally I agree with this location.

Reply:
The reason it's exported like this is we do gRPC call outside the context of the project all the time, so it needed to be exported. If it were an accessor, I'd still want a code to wrap the metadata addition functionality so that it's not sprinkled everywhere we setup a context to do an RPC call.

Reply:
I don't think putting a context wrapper on the server package is the right place, I'm not sure you'd want to import it everywhere that you do an RPC call.

Reply:
Opt'd to basically do what you said originally given the refactoring into a grpcmetadata package clean it up otherwise.